src/net/WebRTC/RTCPeerConnection.cs

```diff
@@ -327,8 +341,12 @@ public RTCCertificate CurrentCertificate
             // No certificate was provided so create a new self signed one.
             if (_configuration.certificates == null || _configuration.certificates.Count == 0)
             {
-                _currentCertificate = new RTCCertificate { Certificate = DtlsUtils.CreateSelfSignedCert() };
-                _configuration.certificates = new List<RTCCertificate> { _currentCertificate };
+                // Since it's a self signed certificate, reuse it
+                if (_currentCertificate == null)
+                {
+                    _currentCertificate = new RTCCertificate { Certificate = SelfSignedCert };
+                    _configuration.certificates = new List<RTCCertificate> { _currentCertificate };
+                }

```

Comment:
This makes an assumption that an application wants to use the same certificate on every peer connection. This raises security and privacy concerns. If an application developer does want to do that they can create a certificate using `DtlsUtils.CreateSelfSignedCert`, store it and then pass it in as a parameter on the `configuration` object.

The certificate re-use should not be forced on the application.

Reply:
This is only for self-signed certificates, which should not be used in a production environment anyway. Regenerating a self signed certificate is expensive and not necessary.

Reply:
I'm pretty sure Chrome, FireFox etc. all use self-signed certificates by default. Amongst others one reason to regenerate them is privacy. If I'm able to snoop traffic at some upstream router I would be able to link repeated uses of the same certificate to a single user. General rule of thumb is to follow `libwebrtc` and it generates a unique certificate for every peer connection.

Reply:
So... basically self signed certificates are OK?

Reply:
Yes. You'll note when you use Chrome etc. you never get any information about the DTLS certificate. By default the certificate is used to prevent MITM attacks and does not provide any identity claims.

Reply:
OK - I will then make this an opt-in for cert reuse like you mentioned in the first comment. (which is espcially useful for testing on a larger number of connections)